main.go

```diff
@@ -149,131 +150,124 @@ func initDb(ctx context.Context, tx *sqlx.Tx) error {
+		product, err := productsmodel.GetBestMatch(ctx, tx, source.String(), source.Start)
+		if err != nil {
+			return err
+		}
 
-		// Ensure a suitable dateTime object
-		dateTime := datetimesmodel.New(bucketMetricsData.TimeSeries[0].Start)
-		dateTime, err = datetimesmodel.Ensure(ctx, tx, dateTime)
-		checkErrExit(err)
-
-		if bucketMetricsData.TimeSeries[0].Usage.StorageBytes > 0 {
-			fmt.Printf("syncing %s\n", sourceStorage)
-			product, err := productsmodel.GetBestMatch(ctx, tx, sourceStorage, bucketMetricsData.TimeSeries[0].Start)
-			checkErrExit(err)
-			discount, err := discountsmodel.GetBestMatch(ctx, tx, sourceStorage, bucketMetricsData.TimeSeries[0].Start)
-			checkErrExit(err)
-			query, err := queriesmodel.GetByName(ctx, tx, sourceQueryStorage+":"+zone)
-			checkErrExit(err)
-			storageQuantity := float64(bucketMetricsData.TimeSeries[0].Usage.StorageBytes) / 1000 / 1000 / 1000
-			storageFact := factsmodel.New(dateTime, query, tenant, category, product, discount, storageQuantity)
-			_, err = factsmodel.Ensure(ctx, tx, storageFact)
-			checkErrExit(err)
+		discount, err := discountsmodel.GetBestMatch(ctx, tx, source.String(), source.Start)
+		if err != nil {
+			return err
 		}
 
-		if bucketMetricsData.TimeSeries[0].Usage.SentBytes > 0 {
-			fmt.Printf("syncing %s\n", sourceTrafficOut)
-			product, err := productsmodel.GetBestMatch(ctx, tx, sourceTrafficOut, bucketMetricsData.TimeSeries[0].Start)
-			checkErrExit(err)
-			discount, err := discountsmodel.GetBestMatch(ctx, tx, sourceTrafficOut, bucketMetricsData.TimeSeries[0].Start)
-			checkErrExit(err)
-			query, err := queriesmodel.GetByName(ctx, tx, sourceQueryTrafficOut+":"+zone)
-			checkErrExit(err)
-			trafficOutQuantity := float64(bucketMetricsData.TimeSeries[0].Usage.SentBytes) / 1000 / 1000 / 1000
-			trafficOutFact := factsmodel.New(dateTime, query, tenant, category, product, discount, trafficOutQuantity)
-			_, err = factsmodel.Ensure(ctx, tx, trafficOutFact)
-			checkErrExit(err)
+		query, err := queriesmodel.GetByName(ctx, tx, source.Query+":"+source.Zone)
+		if err != nil {
+			return err
 		}
 
-		if bucketMetricsData.TimeSeries[0].Usage.Requests > 0 {
-			fmt.Printf("syncing %s\n", sourceTrafficOut)
-			product, err := productsmodel.GetBestMatch(ctx, tx, sourceRequests, bucketMetricsData.TimeSeries[0].Start)
-			checkErrExit(err)
-			discount, err := discountsmodel.GetBestMatch(ctx, tx, sourceRequests, bucketMetricsData.TimeSeries[0].Start)
-			checkErrExit(err)
-			query, err := queriesmodel.GetByName(ctx, tx, sourceQueryRequests+":"+zone)
-			checkErrExit(err)
-			requestsQuantity := float64(bucketMetricsData.TimeSeries[0].Usage.Requests) / 1000
-			requestsFact := factsmodel.New(dateTime, query, tenant, category, product, discount, requestsQuantity)
-			_, err = factsmodel.Ensure(ctx, tx, requestsFact)
-			checkErrExit(err)
+		var quantity float64
+		if query.Unit == "GB" || query.Unit == "GB/day" {
+			quantity = float64(value) / 1000 / 1000 / 1000
```

Comment:
Is cloudscale using base 10 for their calculations, too?

Reply:
Yes, they're using SI units (1000) for S3. In contrast to VMs, they're using factors of 1024 there.